src/analyzer/protocol/rawstream/RawStream.cc

```diff
@@ -0,0 +1,25 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "zeek/analyzer/protocol/rawstream/RawStream.h"
+
+#include "zeek/analyzer/protocol/rawstream/events.bif.h"
+
+namespace zeek::analyzer::rawstream {
+
+RawStream_Analyzer::RawStream_Analyzer(Connection* conn) : analyzer::tcp::TCP_ApplicationAnalyzer("RawStream", conn) {}
+
+
+void RawStream_Analyzer::DeliverStream(int len, const u_char* data, bool orig) {
+    analyzer::tcp::TCP_ApplicationAnalyzer::DeliverStream(len, data, orig);
+
+    auto s = len > 0 ? zeek::make_intrusive<StringVal>(len, reinterpret_cast<const char*>(data)) :
+                       zeek::val_mgr->EmptyString();
+    BifEvent::enqueue_rawstream_deliver(this, Conn(), orig, std::move(s));
+}
+void RawStream_Analyzer::Undelivered(uint64_t seq, int len, bool orig) {
+    analyzer::tcp::TCP_ApplicationAnalyzer::Undelivered(seq, len, orig);
+
+    BifEvent::enqueue_rawstream_undelivered(this, Conn(), orig, seq, len);
```

Comment:
Same here on `len==zero` (not sure if that even happens, but for consistency in case if ever did). Or the other way around: always pass even zero to script-land for both delivered and undelivered.